tracer/src/Datadog.Trace/PDBs/DatadogPdbReader.cs

```diff
@@ -0,0 +1,60 @@
+// <copyright file="DatadogPdbReader.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+using Datadog.Trace.Vendors.dnlib.DotNet;
+using Datadog.Trace.Vendors.dnlib.DotNet.MD;
+using Datadog.Trace.Vendors.dnlib.DotNet.Pdb;
+using Datadog.Trace.Vendors.dnlib.DotNet.Pdb.Symbols;
+using Datadog.Trace.Vendors.dnlib.IO;
+
+namespace Datadog.Trace.PDBs
+{
+    /// <summary>
+    /// Reads both Windows and Portable PDBs.
+    /// Note: reading Windows PDBs is only supported on Windows.
+    /// </summary>
+    internal class DatadogPdbReader : IDisposable
+    {
+        private readonly SymbolReader _symbolReader;
+        private readonly ModuleDefMD _module;
+
+        private DatadogPdbReader(SymbolReader symbolReader, ModuleDefMD module)
+        {
+            _symbolReader = symbolReader;
+            _module = module;
+        }
+
+        public static DatadogPdbReader CreatePdbReader(string assemblyFullPath)
+        {
+            var module = ModuleDefMD.Load(File.OpenRead(assemblyFullPath));
```

Comment:
According to the documentation of `ModuleDefMD`, the stream is owned by the caller, so you need to dispose it.

Reply:
Great catch, thank you! Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2592/commits/303ff50854b5271c40bfcd2cd03b13050f624b7c